server/src/main/scala/controllers/application/ShiftsController.scala

```diff
@@ -122,6 +137,42 @@ class ShiftsController @Inject()(cc: ControllerComponents,
     }.getOrElse(Future.successful(BadRequest("Expecting JSON data")))
   }
 
+  def getShiftsWithStaffAssignments(port: String, terminal: String, viewDate: String, interval: Int, dayRange: String): Action[AnyContent] = Action.async {
+    ctrl.shiftsService.getShifts(port, terminal).map { shifts =>
+      staffAssignmentsService.allStaffAssignments.map { staffAssignments =>
+        shiftsWithStaffAssignments(staffAssignments, terminal, viewDate, interval, dayRange, shifts)
+      }
+    }.flatten
+  }
+
+  private def shiftsWithStaffAssignments(staffAssignments: ShiftAssignments,
+                                         terminal: String,
+                                         viewDate: String,
+                                         interval: Int,
+                                         dayRange: String,
+                                         shifts: Seq[Shift]) = {
+    val shiftSummaryStaffing: Seq[ShiftSummaryStaffing] = ShiftsHelpers.generateShiftSummaries(SDate(viewDate),
+      dayRange,
+      Terminal(terminal),
+      shifts,
+      staffAssignments,
+      interval)
+    Ok(shiftSummaryStaffing.toJson.compactPrint)
+  }
+
+  def saveShiftsWithStaffAssignments(port: String, terminal: String, viewDate: String, interval: Int, dayRange: String): Action[AnyContent] = Action.async {
```

Comment:
This function doesn't save anything as far as I can tell. Why does it have this name?